### PR TITLE
PR: Migrate IPython Console to new API

### PR DESCRIPTION
**Pull request by goanpeca**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes #

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

**Comment by goanpeca** [comment]
@impact27 @ccordoba12 I am having some test that still fail, but I am not sure why. Could you give a hand here. I am a bit lost since most of it pertains to debugging.

The tests are

```python
# pytest spyder/plugins/ipythonconsole/tests/test_ipythonconsole.py::test_pdb_multiline --run-slow
# pytest spyder/app/tests/test_mainwindow.py::test_move_to_first_breakpoint --run-slow
# pytest spyder/app/tests/test_mainwindow.py::test_set_new_breakpoints --run-slow
# pytest spyder/app/tests/test_mainwindow.py::test_report_comms_error --run-slow
# pytest spyder/app/tests/test_mainwindow.py::test_break_while_running --run-slow
# pytest spyder/app/tests/test_mainwindow.py::test_path_manager_updates_clients --run-slow
# pytest spyder/app/tests/test_mainwindow.py::test_running_namespace --run-slow
# pytest spyder/app/tests/test_mainwindow.py::test_tight_layout_option_for_inline_plot --run-slow
```

**Comment by impact27** [comment]
> ~@impact27 @ccordoba12 I am having some tests that still fail, but I am not sure why. Could you give a hand here? I am a bit lost since most of it pertains to debugging.~
> 
> The tests are:
> 
> ```python
> # pytest spyder/plugins/ipythonconsole/tests/test_ipythonconsole.py::test_pdb_multiline --run-slow
> ```
> 
> Thanks

Is that still a problem? It would be helpful to get the error message to know why it fails, but the tests are passing?

**Comment by goanpeca** [comment]
Yeah @impact27 I thinkg I found the issues so all good

**Comment by goanpeca** [comment]
Hi @dalthviz I still need to fix refactor the code a bit. But in theory, you could start doing QA testing. Thanks a lot!

**Comment by ccordoba12** [comment]
This has several conflicts too. Please update it.

**Review by impact27 — COMMENTED** [review]
I ran into an issue while using remote kernel where the variable explorer doesn't update when using a non-spyder kernel. Because the communication is changing in spyder 5, it should be fixed here. For reference, see the changes I would have done for Spyder 4.x:
https://github.com/spyder-ide/spyder/commit/22d7cb798797966c6cf290a006eaf03c45420db1

**Comment by ccordoba12** [comment]
Hey @impact27, we plan to finish this one in a month or so. But if it's urgent for you, perhaps you could give us a hand to do that?

I think @goanpeca would be happy to give you some pointers on how to do that.

**Comment by impact27** [comment]
I don’t really use non-spyder kernels console so I don’t need that fixed, but let me know if I can help.

File: spyder/plugins/ipythonconsole/widgets/main_widget.py [hunk 1]
```diff
@@ -0,0 +1,1971 @@
...
+        """Update actions following the execution state of the kernel."""
+        client = self.get_current_client()
+        if client is not None:
+            executing = client.stop_button.isEnabled()
+            self.interrupt_action.setEnabled(executing)
+
+    def connect_external_kernel(self, shellwidget):
+        """
+        Connect an external kernel to the Variable Explorer, Help and
+        Plots, but only if it is a Spyder kernel.
```
[the remark this answers is not in the document]
Comment (goanpeca) [reply spyder/plugins/ipythonconsole/widgets/main_widget.py:1629]: Probably :)

File: spyder/plugins/ipythonconsole/widgets/main_widget.py [hunk 2]
```diff
@@ -0,0 +1,1971 @@
...
+        """
+        sw = shellwidget
+        kc = shellwidget.kernel_client
+
+        self.sig_shellwidget_process_started.emit(sw)
+        kc.stopped_channels.connect(
+            lambda sw=sw: self.sig_shellwidget_process_finished.emit(sw))
+        sw.set_namespace_view_settings()
+        sw.refresh_namespacebrowser()
```
[the remark this answers is not in the document]
Comment (goanpeca) [reply spyder/plugins/ipythonconsole/widgets/main_widget.py:1638]: Not sure, cannot recall this now.

File: spyder/plugins/ipythonconsole/widgets/main_widget.py [hunk 3]
```diff
@@ -0,0 +1,1971 @@
...
+    """
+    FIXME:
+    """
+
+    sig_exception_occurred = Signal(dict)
+    """
+    FIXME:
+    """
+
+    sig_shellwidget_process_started = Signal(object)
```
[the remark this answers is not in the document]
Comment (goanpeca) [reply spyder/plugins/ipythonconsole/widgets/main_widget.py:208]: Hi @impact27, thanks for the suggestions. From some docs I started writing

Signals should be named in lowercase and start with the `sig_` prefix.

If signals inform of something that was made or accomplished they should
use the verb in past as the last part of the signals. Example:

`sig_file_deleted` -> The widget that emits this executes the action and
informs this execution to any other widgets connected to this signal

If signals are requesting to execute an action, signals should then use the
suffix `_requested`. Examples:

`sig_delete_file_requested` -> The widget that emits this cannot execute the
action but informs the request for it to be executed by any other widget
connected to this signal

---

So based on that, if process is not the right word, we need to change it, but keeping in mind consistency. `sig_new_shellwidget` does not follow that.

File: spyder/plugins/ipythonconsole/widgets/main_widget.py [hunk 4]
```diff
@@ -0,0 +1,1971 @@
...
+    """
+    This signal is emitted when a shellwidget process starts.
+
+    Parameters
+    ----------
+    shellwidget: spyder.plugins.ipyconsole.widgets.shell.ShellWidget
+        The shellwigdet.
+    """
+
+    sig_shellwidget_process_finished = Signal(object)
```
[the remark this answers is not in the document]
Comment (goanpeca) [reply spyder/plugins/ipythonconsole/widgets/main_widget.py:218]: Idem, see https://github.com/spyder-ide/spyder/pull/12749#discussion_r531753029

**Comment by ccordoba12** [comment]
Closing in favor of #16324

**Pull request closed without merging**